tests/test_defects.py

```diff
@@ -103,30 +131,442 @@ def setUp(self):
+        self.allDefectsList = measAlg.Defects()
+
+        self.brightDefectsList = measAlg.Defects()
+        for d in self.brightBBoxes:
+            self.brightDefectsList.append(d)
+            self.allDefectsList.append(d)
+
+        self.darkDefectsList = measAlg.Defects()
+        for d in self.darkBBoxes:
+            self.darkDefectsList.append(d)
+            self.allDefectsList.append(d)
+
+    def test_maskBlocksIfIntermitentBadPixelsInColumnTest1(self):
+        # continuous bad column
+
+        config = copy.copy(self.defaultConfig)
+        config.badOnAndOffPixelColumnThreshold = 10
+        config.goodPixelColumnGapThreshold = 5
+        config.nPixBorderUpDown = 0
+        config.nPixBorderLeftRight = 0
+
+        task = cpPipe.defects.FindDefectsTask(config=config)
+
+        expectedDefects = [Box2I(corner = Point2I(15, 1), dimensions = Extent2I(1, 50))]
+        defects = self.allDefectsList
+
+        defectsWithColumns = task.maskBlocksIfIntermitentBadPixelsInColumn(defects)
+
+        boxesMeasured = []
+        for defect in defectsWithColumns:
+            boxesMeasured.append(defect.getBBox())
+
+        for boxInput in expectedDefects:
+            self.assertIn(boxInput, boxesMeasured)
+
+    def test_maskBlocksIfIntermitentBadPixelsInColumnTest2(self):
+        # Test 2: n contiguous bad pixels in a column where n >= threshold
+
+        config = copy.copy(self.defaultConfig)
+        config.badOnAndOffPixelColumnThreshold = 10
+        config.goodPixelColumnGapThreshold = 5
+        config.nPixBorderUpDown = 0
+        config.nPixBorderLeftRight = 0
+
+        task = cpPipe.defects.FindDefectsTask(config=config)
+
+        expectedDefects = [Box2I(corner = Point2I(20, 1), dimensions = Extent2I(1, 25))]
+        defects = self.allDefectsList
+
+        defectsWithColumns = task.maskBlocksIfIntermitentBadPixelsInColumn(defects)
+
+        boxesMeasured = []
+        for defect in defectsWithColumns:
+            boxesMeasured.append(defect.getBBox())
+
+        for boxInput in expectedDefects:
+            self.assertIn(boxInput, boxesMeasured)
+
+    def test_maskBlocksIfIntermitentBadPixelsInColumnTest3(self):
+        # Test 3: n contiguous bad pixels in a column where n < threshold
+
         config = copy.copy(self.defaultConfig)
+        config.badOnAndOffPixelColumnThreshold = 10
+        config.goodPixelColumnGapThreshold = 5
         config.nPixBorderUpDown = 0
         config.nPixBorderLeftRight = 0
+
+        task = cpPipe.defects.FindDefectsTask(config=config)
+
+        expectedDefects = [Box2I(corner = Point2I(25, 1), dimensions = Extent2I(1, 8))]
+        defects = self.allDefectsList
+
+        defectsWithColumns = task.maskBlocksIfIntermitentBadPixelsInColumn(defects)
+
+        boxesMeasured = []
+        for defect in defectsWithColumns:
+            boxesMeasured.append(defect.getBBox())
+
+        for boxInput in expectedDefects:
+            self.assertIn(boxInput, boxesMeasured)
```

Comment:
You could clean this up quite a bit by having a `check_maskBlocks` method that takes the input bad data and the expected defects, and then creates the image, calls `maskBlocks` and does the test. Then each individual test would look like this:

```
test_maskBlocks_long_column(self):
    badPixels = [(1,15,1,50), ...]
    expected = [Box2I(...)]
    check_maskBlocks(badPixels, expected)
```